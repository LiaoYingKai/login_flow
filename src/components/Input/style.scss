.input {
	&__content {
		background-color: #fff;
		display: flex;
		align-items: center;
		padding: 8px 12px;
		border-radius: 8px;
	}
	input {
		border: none;
		border-bottom: 1px solid #f8f8f8;
		height: 24px;
		flex: 1;
		margin-left: 8px;
		font-size: 18px;
	}
	> p {
		color: red;
		text-align: left;
		margin: 4px;
	}
}