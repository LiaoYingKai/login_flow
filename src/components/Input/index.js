import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './style.scss';

const propTypes = {
	className: PropTypes.string,
	onChange: PropTypes.func,
	value: PropTypes.string,
	isDisable: PropTypes.bool,
	isReadOnly: PropTypes.bool,
	placeholder: PropTypes.string,
	prefixItem: PropTypes.any,
	// verification: PropTypes.arrayOf(PropTypes.shape({
	// 	rule: PropTypes.string,
	// 	VerificationText: PropTypes.string,
	// })),
	type: PropTypes.string,
};

const defaultProps = {
	onChange: () => {},
	isDisable: false,
	isReadOnly: false,
	verification: [],
};

function Input({
	className,
	onChange,
	value,
	isDisable,
	isReadOnly,
	placeholder,
	prefixItem,
	// verification,
	type,
}) {
	// const isAllVerification = verification.filter(item => {
	// 	if (value.test(item.rule)) {
	// 		return item.VerificationText;
	// 	}
	// });

	// const verificationText = isAllVerification[0] ? isAllVerification[0] : false;

	return (
		<div className={cx('input', className)}>
			<div className="input__content">
				{prefixItem}
				<input
					onChange={onChange}
					value={value}
					readOnly={isReadOnly}
					disabled={isDisable}
					placeholder={placeholder}
					type={type}
				/>
			</div>
			<p>
				{/* { verificationText || null } */}
				{/* test */}
			</p>
		</div>
	);
}

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;

export default Input;
