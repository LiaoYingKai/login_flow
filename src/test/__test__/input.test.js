/* eslint-disable no-undef */
import React from 'react';
import Input from '../../components/Input';
import { shallow, mount, } from "enzyme";
import toJson from 'enzyme-to-json';

describe("Input", () => {
	it('should handle default ptops', () => {
		const {
			onChange,
			isDisable,
			isReadOnly,
		} = Input.defaultProps;

		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
		expect(isDisable).toBe(false);
		expect(isReadOnly).toBe(false);
	});

	it('should handle onChange event', () => {
		const onChange = jest.fn();
		const wrapper = shallow(<Input onChange={onChange} />);

		wrapper.find('input').at(0).simulate('change', { target: { value: 'mock-value' } });
		expect(onChange).toHaveBeenCalledTimes(1);
		expect(onChange).toHaveBeenCalled();
	});
	it("should render corrent", () => {
		const className = 'mock-class';
		const onChange = () => {};
		const value = 'mock-value';
		const isDisable = false;
		const isReadOnly = false;
		const placeholder = 'mock-placeholder';
		const prefixItem = '';
		const errorText = '';
		const type = ''; 

		const wrapper = shallow(<Input
			className={className}
			onChange={onChange}
			value={value}
			isDisable={isDisable}
			isReadOnly={isReadOnly}
			placeholder={placeholder}
			prefixItem={prefixItem}
			errorText={errorText}
			type={type}
		/>);

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should mount in a full DOM", () => {
		const className = 'mock-class';
		const onChange = () => {};
		const value = 'mock-value';
		const isDisable = false;
		const isReadOnly = false;
		const placeholder = 'mock-placeholder';
		const prefixItem = '';
		const errorText = '';
		const type = ''; 

		const wrapper = mount(<Input
			className={className}
			onChange={onChange}
			value={value}
			isDisable={isDisable}
			isReadOnly={isReadOnly}
			placeholder={placeholder}
			prefixItem={prefixItem}
			errorText={errorText}
			type={type}
		/>);

		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().onChange).toEqual(onChange);
		expect(wrapper.props().value).toEqual(value);
		expect(wrapper.props().isDisable).toEqual(isDisable);
		expect(wrapper.props().isReadOnly).toEqual(isReadOnly);
		expect(wrapper.props().placeholder).toEqual(placeholder);
		expect(wrapper.props().prefixItem).toEqual(prefixItem);
		expect(wrapper.props().errorText).toEqual(errorText);
		expect(wrapper.props().type).toEqual(type);
	});
});

describe('Input Readonly type', () => {
	it('should change isReadonly is true', () => {
		const isReadOnly = true;
		const wrapper = mount(<Input isReadOnly={isReadOnly}/>);

		expect(wrapper.props().isReadOnly).toEqual(isReadOnly);
	});
	it('should change isReadonly is false', () => {
		const isReadOnly = false;
		const wrapper = mount(<Input isReadOnly={isReadOnly}/>);

		expect(wrapper.props().isReadOnly).toEqual(isReadOnly);
	});
});

describe('Input Disabled type', () => {
	it('should change isDisabled is true', () => {
		const isDisabled = true;
		const wrapper = mount(<Input isDisabled={isDisabled}/>);

		expect(wrapper.props().isDisabled).toEqual(isDisabled);
	});
	it('should change isDisabled is false', () => {
		const isDisabled = false;
		const wrapper = mount(<Input isDisabled={isDisabled}/>);

		expect(wrapper.props().isDisabled).toEqual(isDisabled);
	});
});
