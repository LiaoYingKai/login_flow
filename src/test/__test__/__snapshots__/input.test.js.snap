// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render corrent 1`] = `
<div
  className="input mock-class"
>
  <div
    className="input__content"
  >
    <input
      disabled={false}
      onChange={[Function]}
      placeholder="mock-placeholder"
      readOnly={false}
      type=""
      value="mock-value"
    />
  </div>
</div>
`;
